File: 0. Basic-Level-May-2015/[EXAMS] Programming-Basics/CSharp-Basics-Exam-12-April-2014-Morning/01. Triangle/Triangle.cs
using System;

public class Triangle 
{
    public static void Main()
    {
        int aX = int.Parse(Console.ReadLine());
        int aY = int.Parse(Console.ReadLine());
        int bX = int.Parse(Console.ReadLine());
        int bY = int.Parse(Console.ReadLine());
        int cX = int.Parse(Console.ReadLine());
        int cY = int.Parse(Console.ReadLine());

        double ab = Math.Sqrt(
            (bX - aX) * (bX - aX) +
            (bY - aY) * (bY - aY));

        double bc = Math.Sqrt(
            (bX - cX) * (bX - cX) +
            (bY - cY) * (bY - cY));

        double ca = Math.Sqrt(
            (cX - aX) * (cX - aX) +
            (cY - aY) * (cY - aY));

        bool canFormATriangle =
            ab + bc > ca && ab + ca > bc && ca + bc > ab;

        if (!canFormATriangle)
        {
            Console.WriteLine("No");
            Console.WriteLine("{0:F2}", ab);
        }
        else
        {
            Console.WriteLine("Yes");
            double p = (ab + bc + ca) / 2;
            double area = Math.Sqrt(p * (p - ab) * (p - bc) * (p - ca));
            Console.WriteLine("{0:F2}", area);
        }
    }
}

File: 1. Fundamental-Level/CSharp/02. Java-Fundamentals-Octomber-2015/[HW] Java-Fundamentals/5. [HW] OOP-In-Java/Problem1_Geometry/PlaneShapes/Triangle.java
package Problem1_Geometry.PlaneShapes;

import Problem1_Geometry.Vertices.Vertex2D;

public class Triangle extends PlaneShape{
    private Vertex2D a;
    private Vertex2D b;
    private Vertex2D c;

    public Triangle(Vertex2D a, Vertex2D b, Vertex2D c ) {
        super(a, b, c);
        this.a = a;
        this.b = b;
        this.c = c;
        this.area = calculateArea();
        this.perimeter = calculatePerimeter();
    }

    @Override
    public double calculateArea() {
        return Math.abs(((this.a.getX() * (this.b.getY() - this.c.getY()))
                + (this.b.getX() * (this.c.getY() - this.a.getY()))
                + (this.c.getX() * (this.a.getY() - this.b.getY()))) / 2);
    }

    @Override
    public double calculatePerimeter() {
        double sideAB = Math.sqrt(((this.b.getX() - this.a.getX()) * (this.b.getX() - this.a.getX())) + ((this.b.getY() - this.a.getY()) * (this.b.getY() - this.a.getY())));
        double sideBC = Math.sqrt(((this.c.getX() - this.b.getX()) * (this.c.getX() - this.b.getX())) + ((this.c.getY() - this.b.getY()) * (this.c.getY() - this.b.getY())));
        double sideAC = Math.sqrt(((this.c.getX() - this.a.getX()) * (this.c.getX() - this.a.getX())) + ((this.c.getY() - this.a.getY()) * (this.c.getY() - this.a.getY())));
        return sideAB + sideBC + sideAC;
    }

    @Override
    public String toString() {
        return "Triangle{" +
                "a=" + a +
                ", b=" + b +
                ", c=" + c +
                '}';
    }
}